pkg/component/config/genericcomponent.go

```diff
@@ -51,7 +51,7 @@ func NewDefaultGenericComponentConfiguration() GenericComponentConfiguration {
 	leaderElection := componentbaseconfigv1alpha1.LeaderElectionConfiguration{
 		// https://github.com/kubernetes/kubernetes/pull/84084
 		// https://github.com/kubernetes/kubernetes/pull/106852
-		ResourceLock: resourcelock.EndpointsLeasesResourceLock,
+		ResourceLock: resourcelock.LeasesResourceLock,
```

Comment:
change to use `leases` from this version, previous v0.6.0 used `endpointsleases`.

Reply:
Do we need to add more comments to indicate why we migrated `EndpointsLeasesResourceLock` to  `LeasesResourceLock`?

Reply:
updated the comment.

> // Use lease-based leader election to reduce cost.
> // We migrated for EndpointsLease lock in 1.17 and starting in 1.20 we
> // migrated to Lease lock.